snuba/subscriptions/data.py

```diff
@@ -0,0 +1,46 @@
+from dataclasses import dataclass
+from datetime import datetime, timedelta
+from typing import List, Tuple
+
+from snuba.datasets.dataset import Dataset
+from snuba.datasets.factory import get_dataset
+from snuba.views import validate_request_content
+from snuba.query.types import Condition
+from snuba.request import Request
+from snuba.request.schema import RequestSchema
+
+
+@dataclass(frozen=True)
+class Subscription:
+    """
+    Represents the state of a subscription.
+    """
+    id: str
+    project_id: int
+    dataset: str
```

Comment:
I think we'll end up with a subscription topic per dataset that is exclusively used for that dataset. If we do that, this value will be redundant. I'd also be a little hesitant about dynamically referencing datasets as part of `build_request` (though to be honest, I can't give you a good reason why.)

Reply:
One thought here is that the subscription is to a query, and dataset is a core part of a query, since the conditions that can be used are tied to that. I can see that it might not make sense for `build_request` to return it, since it overloads the name of the method.

I guess one concern could be if dataset names change, but I'm not sure whether that's likely?

Overall I don't feel too strongly about keeping it here though.